Bug: can't run a single ESLint rule in CLI after migrating to flat config

### Environment

Node version: 20.12.2
npm version: 10.5.0
Local ESLint version: 8.57.0
Global ESLint version: 8.57.0
Operating System: Ubuntu 22


### What parser are you using?

@typescript-eslint/parser

### What did you do?

I have a pre-commit script that uses the CLI to run a subset of ESLint rules on commit.  A minimal version of the command I'm running:

```
npx eslint --no-eslintrc \
    --parser '@typescript-eslint/parser' \
    --plugin 'import' \
    --plugin 'unused-imports' \
    --rule 'unused-imports/no-unused-imports: [error]' \
    --rule 'import/order: [error, {alphabetize: {order: asc, caseInsensitive: true}}]' \
    --fix \
    $IMPORT_FILES
```

After migrating to flat config, this command fails and tells me to switch to `--no-config-lookup` instead of `--no-eslintrc`. When I make the change, it still fails with "File ignored because of a matching ignore pattern.", likely because of https://github.com/eslint/eslint/issues/18263.

The workaround I'm using in the meantime is to prepend the old command with ESLINT_USE_FLAT_CONFIG=false.

### What did you expect to happen?

After switching to `--no-config-lookup`, the command continues to work.

### What actually happened?

```
  0:0  warning  File ignored because of a matching ignore pattern. Use "--no-ignore" to disable file ignore settings or use "--no-warn-ignored" to suppress this warning

✖ 1 problem (0 errors, 1 warning)
```

### Link to Minimal Reproducible Example

See above

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

_No response_

--- a/docs/src/use/configure/migration-guide.md
+++ b/docs/src/use/configure/migration-guide.md
@@ -674,6 +674,23 @@ You can see the TypeScript types for the flat config file format in the Definite
 
 You can view the type definitions in the [DefinitelyTyped repository on GitHub](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/eslint/index.d.ts).
 
+## Visual Studio Code Support
+
+By default, the Visual Studio Code ESLint plugin does not look for `eslint.config.js` files. To enable support for the new configuration files, edit your `.vscode/settings.json` file and add the following:
+
+```json
+{
+  /// enable flat config files (eslint.config.*js)
+  "eslint.experimental.useFlatConfig": true
+}
+```
+
+In a future version of the ESLint plugin, you will no longer need to enable this manually.
+
+::: important
+The ESLint team does not maintain the ESLint plugin for Visual Studio Code, and as such, we don't have any control over its release cycle. We are currently waiting on [this issue](https://github.com/microsoft/vscode-eslint/issues/1644) to be resolved.
+:::
+
 ## Further Reading
 
 * [Overview of the flat config file format blog post](https://eslint.org/blog/2022/08/new-config-system-part-2/)

--- a/docs/src/use/configure/migration-guide.md
+++ b/docs/src/use/configure/migration-guide.md
@@ -676,7 +676,7 @@ You can view the type definitions in the [DefinitelyTyped repository on GitHub](
 
 ## Visual Studio Code Support
 
-By default, the Visual Studio Code ESLint plugin does not look for `eslint.config.js` files. To enable support for the new configuration files, edit your `.vscode/settings.json` file and add the following:
+By default, the [Visual Studio Code ESLint plugin](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) does not look for `eslint.config.js` files. To enable support for the new configuration files, edit your `.vscode/settings.json` file and add the following:
 
 ```json
 {

--- a/docs/src/use/configure/migration-guide.md
+++ b/docs/src/use/configure/migration-guide.md
@@ -688,7 +688,7 @@ By default, the [Visual Studio Code ESLint plugin](https://marketplace.visualstu
 In a future version of the ESLint plugin, you will no longer need to enable this manually.
 
 ::: important
-The ESLint team does not maintain the ESLint plugin for Visual Studio Code, and as such, we don't have any control over its release cycle. We are currently waiting on [this issue](https://github.com/microsoft/vscode-eslint/issues/1644) to be resolved.
+The ESLint team does not maintain the ESLint plugin for Visual Studio Code, and as such, we don't have any control over its release cycle. We are currently waiting on [this issue](https://github.com/microsoft/vscode-eslint/issues/1644 "Microsoft/vscode-eslint issue: Enable Flat Config by default") to be resolved.
 :::
 
 ## Further Reading

--- a/docs/src/use/configure/migration-guide.md
+++ b/docs/src/use/configure/migration-guide.md
@@ -680,7 +680,7 @@ By default, the [Visual Studio Code ESLint plugin](https://marketplace.visualstu
 
 ```json
 {
-  /// enable flat config files (eslint.config.*js)
+  // enable flat config files (eslint.config.*js)
   "eslint.experimental.useFlatConfig": true
 }
 ```

